### PR TITLE
Refactor RoleManager interface and test

### PR DESCRIPTION
**Pull request by nodece**
Fix: https://github.com/casbin/node-casbin/issues/46

### CODE REVIEW

**Comment by hsluoyz** [comment]
@nodece travis failed, please fix.

**Comment by nodece** [comment]
@hsluoyz New version requires node v7.6.0 or higher for ES2015 and async function support.

**Comment by hsluoyz** [comment]
@nodece if this is the case, please remove Node 6 in the travis.

**Comment by nodece** [comment]
ping @hsluoyz

File: examples/rbac_policy.csv [hunk 1]
```diff
@@ -2,5 +2,4 @@ p, alice, data1, read
 p, bob, data2, write
 p, data2_admin, data2, read
 p, data2_admin, data2, write
-
```
Comment (hsluoyz) [inline examples/rbac_policy.csv]: Do not remove this empty line.

**Comment by hsluoyz** [comment]
lgtm

**Pull request merged**